### PR TITLE
[3/6] filterx: += operator with dict and list literals

### PR DESCRIPTION
**Pull request by alltilla**
We don't have generic dicts or lists in filterx. We have a dict and a list interface, which can be implemented by anyone.

Literal dicts and lists do not make sense in this case, however with a bit of trickery we can make them "generator"s, which use the dict and list interfaces and operate on an actual implementation.

For this we introduce the "+=" operator, which on the left hand side expects an object that is a dict or list implementation, and a "generator" on the right hand side (in the form of literal dict and list for now, but functions will be added soon). The "+=" operator passes the "fillable" to the generator, which then fills it appropriately.

Please note that this is a shallow merge:
  * lists are always appended
  * dict merging works like python's dict update() function

Depends on #4860 and #4861

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by kira-syslogng** [comment]
Build FAILURE

File: lib/cfg-lex.l [hunk 1]
```diff
@@ -314,6 +314,7 @@ word	[^ \#'"\(\)\{\}\[\]\\;\r\n\t,|\.@:]
 <INITIAL,filterx>>         { return KW_TA_GT; }
 <INITIAL,filterx>===       { return KW_TAV_EQ; }
 <INITIAL,filterx>!==       { return KW_TAV_NE; }
+<INITIAL,filterx><<        { return KW_SHIFT_LEFT; }
```
Comment (MrAnno) [inline lib/cfg-lex.l:317]: Just wondering, why don't we use something like `+=` and allow `+` too?

Looking at other languages, the "list append" and "dict extend" functionalities are usually implemented as functions or methods, but binary operators like `<<`, `+=`, etc. look better and feel intuitive in my opinion.

Comment (alltilla) [reply lib/cfg-lex.l]: Really good idea! Thanks :)

After talking about this IRL, I have changed this PR to use "+=", but I have a branch in progress which implements "=" and type cast wrapping "=", etc...

I think this PR is useful even without all the operator supports, so I think we can merge this as is, and I will open a new PR with the rest later.

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by MrAnno** [comment]
@kira-syslogng retest this please

File: lib/cfg-lex.l (hunk 1 above)
Comment (MrAnno) [reply lib/cfg-lex.l]: Thanks.

**Review by MrAnno — APPROVED** [review]

**Pull request merged**